QEfficient/src/_transformers/runtime_args.py

```diff
@@ -0,0 +1,33 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+import os
+from dataclasses import dataclass, field
+from typing import List
+
+from QEfficient.utils.constants import Constants
+
+
+@dataclass
+class QEFFAutoModelForCausalLMAI100RuntimeArgs:
```

Comment:
Do we need a new dataclass for runtime args? A simple dict won't work?

Reply:
1. Dataclasses automatically does type restriction of attributes
2. We are doing some check on them in __post_init__ which we will  have to write everywhere they are being initialized if we use dicts
3. Also, I am thinking of freezing this dataclasses so that, user can't assign arbitrary values to them
4. So, because of 2 and 3 we are always sure that once runtime-args are populated we can always allowed to execute in respective runtime, making it robust from failures.

I understand that dicts are more efficient for initialization and access, but I feel the gain is negligible as compared to the above advantages we are getting by using dataclass

ref: 
https://stackoverflow.com/questions/55248416/python-3-which-one-is-faster-for-accessing-data-dataclasses-or-dictionaries
https://stackoverflow.com/questions/74117873/data-classes-vs-dictionaries

Reply:
As suggested above, if we use `execute_` method with correct args or kwargs, that will work as validation. We're going to use it at only one place, for execution. `dataclass` makes more sense when we're re-using in multiple places, serializing or sending over network etc.